tests/telemetry/test_telemetry_metrics.py

```diff
@@ -0,0 +1,255 @@
+            "common": True,
+            "host": ANY,
+            "interval": 60,
+            "name": "dd.app_telemetry.tracers.test-metric",
+            "points": [[1642544540, 2.0], [1642544540, 3.0]],
+            "tags": {"a": "b"},
+            "type": "count",
+        },
+    ]
+
+    _assert_metric(test_agent_session_telemetry_metrics, expected_series)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 6), reason="mock.ANY doesn't works in py3.5 or lower")
+def test_send_metric_datapoint_equal_type_different_tags_expected_3_serie(
+    test_agent_session_telemetry_metrics, mock_time
+):
+    """A datapoint is at least: a metric name, a metric value, and the time at which the value was collected.
+    But in Datadog, a datapoint also includes tags, which declare all the various scopes the datapoint belongs to
+    https://www.datadoghq.com/blog/the-power-of-tagged-metrics/#whats-a-metric-tag
+    """
+    telemetry_writer = test_agent_session_telemetry_metrics.telemetry_writer
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 4, {"a": "b"})
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 5, {"a": "b", "c": "d"})
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 6, {})
+
+    expected_series = [
+        {
+            "common": True,
+            "host": ANY,
+            "interval": 60,
+            "name": "dd.app_telemetry.tracers.test-metric",
+            "points": [[1642544540, 4.0]],
+            "tags": {"a": "b"},
+            "type": "count",
+        },
+        {
+            "common": True,
+            "host": ANY,
+            "interval": 60,
+            "name": "dd.app_telemetry.tracers.test-metric",
+            "points": [[1642544540, 5.0]],
+            "tags": {"a": "b", "c": "d"},
+            "type": "count",
+        },
+        {
+            "common": True,
+            "host": ANY,
+            "interval": 60,
+            "name": "dd.app_telemetry.tracers.test-metric",
+            "points": [[1642544540, 6.0]],
+            "tags": {},
+            "type": "count",
+        },
+    ]
+
+    _assert_metric(test_agent_session_telemetry_metrics, expected_series)
+
+
+def test_send_metric_datapoint_equal_tags_different_type_expected_1_serie(
+    test_agent_session_telemetry_metrics, mock_time
+):
+    """A datapoint is at least: a metric name, a metric value, and the time at which the value was collected.
+    But in Datadog, a datapoint also includes tags, which declare all the various scopes the datapoint belongs to
+    https://www.datadoghq.com/blog/the-power-of-tagged-metrics/#whats-a-metric-tag
+    """
+    telemetry_writer = test_agent_session_telemetry_metrics.telemetry_writer
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 1, {"a": "b"})
+    with pytest.raises(TelemetryTypeError) as e:
+        telemetry_writer.add_gauge_metric(TELEMETRY_TRACER, "test-metric", 1, {"a": "b"})
+
+    assert e.value.args[0] == (
```

Comment:
I think this assertion needs to be within the `with` block to see the `e` variable. It's at least slightly clearer to the reader that way.